libcudacxx/include/cuda/__memcpy_async/cp_async_bulk_shared_global.h

```diff
@@ -37,16 +37,50 @@
 
 #    include <cuda/std/__cccl/prologue.h>
 
+namespace cooperative_groups
+{
+namespace __v1
+{
+class thread_block;
+
+template <unsigned int Size, typename ParentT>
+class thread_block_tile;
+} // namespace __v1
+using namespace __v1;
+} // namespace cooperative_groups
+
 _CCCL_BEGIN_NAMESPACE_CUDA
 
+template <typename _Group>
+[[nodiscard]] _CCCL_DEVICE _CCCL_FORCEINLINE unsigned int __thread_rank(const _Group& __g)
+{
+  return __g.thread_rank();
+}
```

Comment:
Is that a leftover from an earlier version or intended to replace the calculation of `__tid` below?

Reply:
yes, removed.